### PR TITLE
zcash_client_sqlite: Fix more regtest breakage

### LINKED ISSUE #2202: zcash_client_sqlite: Fix more regtest breakage
https://github.com/zcash/librustzcash/blob/707959e0d4b5eb203f86a658378035bdde7202d4/zcash_client_sqlite/src/wallet.rs#L1976-L1978
https://github.com/zcash/librustzcash/blob/707959e0d4b5eb203f86a658378035bdde7202d4/zcash_client_sqlite/src/wallet.rs#L2001-L2003

### PR DESCRIPTION
**Pull request by str4d**
Closes #2202.
Closes COR-1076.

### CODE REVIEW

File: zcash_client_sqlite/src/wallet/init/migrations/v_sapling_shard_unscanned_ranges.rs [hunk 1]
```diff
@@ -64,11 +64,12 @@ impl<P: consensus::Parameters> RusqliteMigration for Migration<P> {
...
-                self.params
-                    .activation_height(NetworkUpgrade::Sapling)
-                    .unwrap()
-            ),
+            // Sapling might not be active in regtest mode.
+            self.params
+                .activation_height(NetworkUpgrade::Sapling)
+                .map(|h| u32::from(h).to_string())
+                .as_deref()
+                .unwrap_or("NULL"),
```
Comment (nuttycom) [inline zcash_client_sqlite/src/wallet/init/migrations/v_sapling_shard_unscanned_ranges.rs:72]: This is a bit weird, in that the substitution is occurring inside a `CREATE VIEW`, meaning that the null value ends up hardcoded within the view, and therefore in any case where `prev_shard.subtree_end_height IS NULL` the view will return no rows. I haven't been able to deeply analyze the implications of this, but it seems like it could create some very weird boundary conditions if Sapling is activated after the test begins. I expect there may be bugs with respect to scanning the initial shard that result.

Comment (str4d) [reply zcash_client_sqlite/src/wallet/init/migrations/v_sapling_shard_unscanned_ranges.rs:72]: In general this shouldn't be a problem, because most tests were updated in `zcashd` to assume a baseline of Sapling a few years ago (and currently `zebrad` forces everything up to Canopy to be activated in regtest at height 1, but that could change). I don't actually recall whether or not there are tests that are solely pre-Sapling, or that activate Sapling after starting.

In theory we could have just left the Sapling assertions alone, and only fixed the NU5 ones. But I prefer us having consistent regtest behaviour across the codebase (and also some of these Sapling assertions would absolutely have broken Zallet trying to sync from genesis rather than from Sapling activation - https://github.com/zcash/wallet/issues/266), so IMO better to just change them all and then work around any specific regtest issues it results in.

Comment (nuttycom) [reply zcash_client_sqlite/src/wallet/init/migrations/v_sapling_shard_unscanned_ranges.rs:72]: Maybe this pitfall should be documented in `wallet/db.rs` on the view documentation so that if there is weirdness in testing in the future, the source of that weirdness is not entirely opaque?

Comment (str4d) [reply zcash_client_sqlite/src/wallet/init/migrations/v_sapling_shard_unscanned_ranges.rs:72]: And the other factor: we do not support changing networks across databases, so this issue won't affect mainnet or testnet databases (created with params that have always had Sapling activation), and regtest databases being repurposed for mainnet and testnet is unsupported and likely breaks in a bunch of other ways.

**Review by nuttycom — APPROVED** [review]
utACK 1fd7966b33b96e3486309e303568f71f2d9746b4
